addons/arrays/fnc_shuffle.sqf

```diff
@@ -14,38 +15,32 @@ Example:
     (begin example)
         _result = [[1, 2, 3, 4, 5]] call CBA_fnc_shuffle;
         // _result could be [4, 2, 5, 1, 3]
-    (end)
 
-TODO:
-    Allow shuffling of elements in-place, using the original array.
+        _array = [1, 2, 3, 4, 5];
+        [_array,true] call CBA_fnc_shuffle;
+        // _array could now be [4, 2, 5, 1, 3]
+    (end)
 
 Author:
     toadlife (version 1.01) http://toadlife.net
-    (rewritten by Spooner)
+    rewritten by Spooner, Dorbedo
 ---------------------------------------------------------------------------- */
 
 #include "script_component.hpp"
 SCRIPT(shuffle);
 
-private ["_shuffledArray", "_tempArray", "_indexToRemove"];
-_shuffledArray = [];
-
-// Support the deprecated parameter style: [1, 2, 3, 4, 5] call CBA_fnc_shuffle.
-_tempArray = if (count _this != 1) then {
-    WARNING("CBA_fnc_shuffle requires an array as first parameter, not just a direct array: " + str _this);
-    [] + _this;
-} else {
-    if (IS_ARRAY(_this select 0)) then {
-        [] + (_this select 0); // Correct params passed.
-    } else {
-        WARNING("CBA_fnc_shuffle requires an array as first parameter, not just a direct array: " + str _this);
-        [] + _this;
-    };
-};
+_this params [["_array",[],[[]]],["_inPlace",false,[false]]];
```

Comment:
The `_this` is superfluous as it's the variable used by the default unary syntax of `params`.